account-decoder/src/lib.rs

```diff
@@ -58,6 +61,22 @@ pub enum UiAccountEncoding {
 
 impl UiAccount {
     pub fn encode(
+        pubkey: &Pubkey,
+        account: AccountNoData,
```

Comment:
Is it possible to take a reference to a `AnAccount` here to generalize over both `AccountNoData` and `Account`?

Reply:
These are the kind of replication that I had mentioned to you in slack I wasn't happy with yet. I figured I'd revisit things like this now that the whole ci is happy. I would love to combine all these _legacy or _no_data functions as much as it makes sense. I'm open to ideas or rust ways to accomplish some of this. I haven't become an expert in rust alternatives to specialization yet. I'm also not an expert in serialization/deserialization yet, and I'm trying to keep things simple. So, there are a few tradeoffs and a few areas where I'm unsure how to tradeoff cost/benefit. In short, yes, it is worth trying to reduce all of these to reduce test burden, reduce duplicated code, etc. But, we also have to consider execution efficiency, of course. Please do provide feedback and ideas. Thanks!

Reply:
yes, and I did that and realized that nobody called encode_legacy anyway. testing that now on the ci. But, I saved the AnAccount version anyway ;-)